app/src/main/java/ru/prometeydev/movie/data/adapters/MoviesAdapter.kt

```diff
@@ -60,29 +66,39 @@ class MoviesAdapter(
         private val like = itemView.findViewById<ImageView>(R.id.like_heart)
 
         fun onBind(movie: Movie, clickListener: OnRecyclerItemClicked) {
-            name.text = movie.name
-            genre.text = movie.genre
-            duration.text = context.getString(R.string.movie_time, movie.duration)
-            rating.rating = movie.rating
-            reviewsCount.text = context.getString(R.string.reviews, movie.reviewsCount)
-            ageLimit.text  = context.getString(R.string.age_limit, movie.ageLimit)
-            filmCover.setImageResource(movie.filmCoverDrawable)
-
-            like.setImageResource(
-                    if (movie.hasLike)
-                        R.drawable.ic_like_active
-                    else
-                        R.drawable.ic_like_inactive
-            )
+            name.text = movie.title
+            genre.text = movie.genres.joinToString { it.name }
+            duration.text = context.getString(R.string.movie_time, movie.runtime)
+            rating.rating = movie.ratings.calculateStarsCount()
+            reviewsCount.text = context.getString(R.string.reviews, movie.numberOfRatings)
+            ageLimit.text  = context.getString(R.string.age_limit, movie.minimumAge)
+
+            scope.launch {
```

Comment:
This is dangerous - do you need to cancel scope when you reuse holder?

Reply:
Yeah, I’ll fix it.

Reply:
I decided to remove the use of corrutines here, since it doesn't seem to affect the speed. I will also remove ImageLoader from Coil.